src/internal.cc

```diff
@@ -163,6 +163,8 @@ internalHostname(void)
     return host;
 }
 
+// XXX: This check mishandles cases where the listening port hostname
+//      does not match the request-target hostname.
```

Comment:
This XXX documents the old/existing problem unrelated to this PR workaround and, arguably, to this instance-wide internalHostnameIs() function. I am not against this XXX, but I recommend moving it to the internalHostnameIs() caller.

There are at least two (very different) problems related to the internal request detection code:

1. We do not have enough port-specific checks. We can document this old problem in this PR as discussed above.
2. The new check added by this PR is flawed. We should document _that_ problem. The XXX above is not enough. I have proposed the corresponding XXX text earlier, but I do not insist on any specific wording. That XXX should go where the new `if` statement is.

Reply:
The flaw itself is not new at all. Just more obvious with the new check being added. This whole function does not verify that any of the names actually match the listening port's configured  host:port or defaultsite= settings.
IMO, doing a complex fix just for "localhost" is nowhere near enough to resolve the issue, so I placed it on the function.